docs/source/traits_user_manual/advanced.rst

```diff
@@ -301,6 +301,38 @@ exception, as does attempting to set an attribute that is not one of the three
 defined attributes. In essence, TreeNode behaves like a type-checked data
 structure.
 
+.. index:: HasRequiredTraits class
+
+.. _hasrequiredtraits:
+
+HasRequiredTraits
+'''''''''''''''''
+
+This class builds on the functionality of HasStrictTraits and ensures
+that any object attribute with `required=True` in its metadata must be passed
```

Comment:
reST nit: use double backticks for code snippets. (Single backticks use the Sphinx default role, which isn't what you want here.)